yoticlient.go

```diff
@@ -131,6 +132,8 @@ func getActivityDetails(requester httpRequester, encryptedToken, sdkID string, k
 					parsedTime, err := time.Parse("2006-01-02", string(attribute.Value))
 					if err == nil {
 						result.DateOfBirth = &parsedTime
+					} else {
+						log.Printf("Unable to parse `date_of_birth` value: %q. Error: %q", attribute.Value, err)
 					}
 				case "postal_address":
```

Comment:
On another note, why are we using `switch/case` on profile attribute retrieval? Imo we need some architecture here, otherwise this SDK will be very hard to extend/maintain at some points.

Reply:
Yes, switch/case is not ideal. The architecture here is something we need to address definitely, but is probably beyond the scope of this PR